Components/Sensors/IMUTask.cpp

```diff
@@ -0,0 +1,274 @@
+/**
+  ******************************************************************************
+  * File Name          : IMUTask.cpp
+  *
+  *	Source Info		   : Based on Andromeda V3.31 Implementation
+  *						 Andromeda_V3.31_Legacy/Core/Src/ReadAccelGyroMagnetism.c
+  *
+  * Description        : This file contains constants and functions designed to
+  *                      obtain accurate pressure and temperature readings from
+  *                      the IMU on the flight board.
+  ******************************************************************************
+*/
+
+/* Includes ------------------------------------------------------------------*/
+#include "IMUTask.hpp"
+
+#include "main.h"
+#include "Data.h"
+#include "DebugTask.hpp"
+#include "Task.hpp"
+
+
+/* Macros --------------------------------------------------------------------*/
+
+/* Structs -------------------------------------------------------------------*/
+
+/* Constants -----------------------------------------------------------------*/
+/* Values should not be modified, non-const due to HAL and C++ strictness) ---*/
+constexpr int CMD_TIMEOUT = 150;
+
+#define READ_CMD_MASK 0x80
+#define WRITE_CMD_MASK 0x00
+#define ACCEL_GYRO_MASK 0x00
+#define MAGNETO_MASK 0x40
+
+// Register addresses
+#define G1_CTRL_REGISTER_ADDR 0x10 // CTRL_REG1_G (10h)
+#define XL6_CTRL_REGISTER_ADDR 0x20 // CTRL_REG6_XL (20h)
+#define M3_CTRL_REGISTER_ADDR 0x22 // CTRL_REG3_M (22h)
+#define WHOAMI_REGISTER_ADDR 0x0F  // WHO_AM_I_A/G (Accel/Gyro - expected value is 104)
+#define WHOAMIM_REGISTER_ADDR 0x0F // WHO_AM_I_M (Magnetometer - expected value is 61)
+
+#define GYRO_X_G_LOW_REGISTER_ADDR 0x18
+#define ACCEL_X_LOW_REGISTER_ADDR 0x28
+#define MAGNETO_X_LOW_REGISTER_ADDR 0x28
+
+#define ACCEL_SENSITIVITY 0.732 // Unit is mg/LSB
+#define GYRO_SENSITIVITY 8.75  // Unit is mdps/LSB
+#define MAGENTO_SENSITIVITY 0.14 // Unit is mgauss/LSB
```

Comment:
These aren't very specific. Checking the datasheet [(from sparkfun not st)](https://cdn.sparkfun.com/assets/learn_tutorials/3/7/3/LSM9DS1_Datasheet.pdf), this is set for 16Gs of linear acceleration, 245dps angular rate, 4 gauss magnetic sensitivity. Instead, we should have consts for the full scales (which mean something):
` constexpr uint16_t xl_full_scale_g = 8`
` constexpr uint16_t gy_full_scale_dps = 2000` // 245dps is way too low
` constexpr uint16_t mg_full_scale_gauss = 4` // I don't know what's appropriate here

and define the sensitivities (that don't mean much). This would mean that elsewhere (in the constructor maybe), we could assign sensitivities depending on the set full scale.
eg:
` (xl_full_scale_g == 2) ? xl_sensitivity = 0.061 : 
  (xl_full_scale_g == 4) ? xl_sensitivity = 0.122 : 
  (xl_full_scale_g == 8) ? xl_sensitivity  = 0.244 :
  (xl_full_scale_g == 16) ? xl_sensitivity = -0.732 :  
  sensitivity = default; `